iceoryx_posh/source/runtime/service_discovery.cpp

```diff
@@ -22,16 +22,18 @@ namespace iox
 namespace runtime
 {
 cxx::expected<ServiceContainer, FindServiceError>
-ServiceDiscovery::findService(const cxx::variant<Wildcard_t, capro::IdString_t> service,
-                              const cxx::variant<Wildcard_t, capro::IdString_t> instance) noexcept
+ServiceDiscovery::findService(const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& service,
+                              const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& instance) noexcept
 {
     /// @todo #415 remove the string mapping, once the find call is done via shared memory
     capro::IdString_t serviceString;
     capro::IdString_t instanceString;
+    bool isServiceWildcard = false;
+    bool isInstanceWildcard = false;
 
     if (service.index() == 0U)
```

Comment:
We could maybe also introduce a helper function 
```
bool isWildCard(const cxx::variant< ... >& value) {
return service.index() == 0U;
}
```

This will be inlined and make the code more expressive.